pkg/crc/machine/bundle/metadata.go

```diff
@@ -237,6 +237,13 @@ func GetBundleNameWithoutExtension(bundleName string) string {
 	return strings.TrimSuffix(bundleName, bundleExtension)
 }
 
+func GetBundleNameWithExtension(bundleName string) string {
+	if strings.HasSuffix(bundleName, bundleExtension) {
+		return bundleName
+	}
+	return fmt.Sprintf("%s.%s", bundleName, bundleExtension)
```

Comment:
`bundleExtension` already has the `.`, it needs to be removed.
And I would add the ` userProvidedBundle = bundle.GetBundleNameWithExtension(userProvidedBundle)` in this commit, and change the commit log to something like:
> Silence incorrect bundle mismatch warning
>
> The previous commit introduced a warning when `crc setup` is used with its `-b` option, and when the bundle version does not correspond to the one hardcoded in crc's Makefile. However, if the bundle is unpacked and one uses `crc setup -b ~/.crc/cache/crc_libvirt_4.11.1_amd64`, the warning will also be printed because `crc_libvirt_4.11.1_amd64` is different from the expected `crc_libvirt_4.11.1_amd64.crcbundle`. While it's questionable if `-b` should accept a path or not, this has been working until now.
> This commit introduces a `GetBundleNameWithExtension()` helper which will be used to silence this warning.

Reply:
@cfergeau This helper function is not silence incorrect bundle mismatch warning but it actually enable to show that warning. Existing commit message have info why we are putting that helper function.

Reply:
> In following commits we are going to get the bundleName from bundle metadata which doesn't have `.crcbundle` extension and this helper provide bundle name with extension to compare with the release bundle version so we can provide the bundle-mismatch warning.

If I go with just this commit log, then imo the whole commit should be squashed in ` Validate: Add warning in case bundle metadata have different bundle name` and `bundle.GetBundleNameWithExtension(userProvidedBundle)` can be moved to `ValidateBundle` to make it clear when it's required to add the extension.

However, this is not the only case when it's needed, if I take the current PR and remove the `userProvidedBundle = bundle.GetBundleNameWithExtension(userProvidedBundle)` line, then `crc setup -b ~/.crc/cache/crc_libvirt_4.11.1_amd64` will print an unwanted warning.
If I checkout `Bundle: Add helper function to get bundle name with extension` and remove the `userProvidedBundle = bundle.GetBundleNameWithExtension(userProvidedBundle)` I also get this warning. With this line added, then the warning is gone.
So at the point you add `Bundle: Add helper function to get bundle name with extension` to the PR, I would say it's fixing an incorrect warning. That it's also useful when you add `GetBundleName` later on is a nice plus, but imo not the main reason we need it.

Reply:
@cfergeau I misunderstood the part of using same bundle but without extension is also create warning for the user and that is an incorrect/unwanted warning and this is what is silence through it. To be honest I didn't thought of it when implemented this part as my initial thinking was just using path always with extension. But now it clear it so updated the commit message.

Reply:
Yes I only realized this morning that you had not noticed this side-effect and added it for a different reason, it's also clear now on my side! :)